espnet2/tasks/lm.py

```diff
@@ -0,0 +1,195 @@
+import argparse
+import logging
+from typing import Any, Dict, Type, Tuple, Optional, Sequence, Callable
+
+import configargparse
+import numpy as np
+import torch
+from typeguard import check_return_type, check_argument_types
+
+from espnet2.lm.abs_model import AbsLM
+from espnet2.lm.e2e import LanguageE2E
```

Comment:
LanguageE2E sounds a bit strange for me.
It might be too later, but `e2e` does not sometimes sound correct.

Reply:
It's not too later because class names and file names  can be easily renamed if using PyCharm, so I roughly name them always.

E2E is not good for LM training, I agree, but I don't have the other idea in my poor vocabulary.
Could you suggest any one?

The role of this class:

 - Delegate: Consists of  some internal blocks. e.g. Encoder, Decoder for ASR, and just a model for LM training.
 - Criterion: Define loss and return it to the Training part.
 - Mediator object between task depended structures and the common region. In other words, any functions in the common training part can't touch any task depended objects except for this class.

My candidates:

- LossModel
- ModelController
- ModelComposer
- ModelMaster
- ESPNetModel -> LanguageESPNetModel, ASRESPNetModel, ...
- Classifier
- E2E

E2E may be good if limiting for ESPNet user because this class has the similar role to the current E2E, but for language model, it sounds strange.